src/cljs/witan/ui/activities.cljs

```diff
@@ -0,0 +1,151 @@
+;; - Where ever an activity includes an event followed by a command, the new command will
+;;   introduce a *new* command ID. At this point the new command has no data connection
+;;   to the previous event so we just cross our fingers and hope for the best. When designing
+;;   activities, be aware of commands that appear in existing activities as this could occurr.
+;;   The code will simply give the new state to the first FSM it comes across that's expecting
+;;   that command, so long as the activity is pending.
+;;   Note: Adjust `extract-command-event-signal` to further select messages
+
+(def available-activities
+  {:upload-file [{:kixi.comms.command/key  :kixi.datastore.filestore/create-upload-link}
+                 {:kixi.comms.event/key    :kixi.datastore.filestore/upload-link-created}
+                 {:kixi.comms.command/key  :kixi.datastore.filestore/create-file-metadata}
+                 (a/or
+                  [{:kixi.comms.event/key  :kixi.datastore.file/created} (a/$ :completed)]
+                  [{:kixi.comms.event/key  :kixi.datastore.file-metadata/rejected} (a/$ :failed)])]
+   ;;
+   :update-metadata [{:kixi.comms.command/key  :kixi.datastore.metadatastore/update}
+                     (a/or
+                      [{:kixi.comms.event/key  :kixi.datastore.file-metadata/updated
+                        :kixi.comms.event/payload {:kixi.datastore.communication-specs/file-metadata-update-type :kixi.datastore.communication-specs/file-metadata-update}}
+                       (a/$ :completed)]
+                      [{:kixi.comms.event/key  :kixi.datastore.metadatastore/update-rejected} (a/$ :failed)])]
+   ;;
+   :update-sharing [{:kixi.comms.command/key  :kixi.datastore.metadatastore/sharing-change}
+                    (a/or
+                     [{:kixi.comms.event/key  :kixi.datastore.file-metadata/updated
+                       :kixi.comms.event/payload {:kixi.datastore.communication-specs/file-metadata-update-type :kixi.datastore.communication-specs/file-metadata-sharing-updated}}
+                      (a/$ :completed)]
+                     [{:kixi.comms.event/key  :kixi.datastore.metadatastore/sharing-change-rejected} (a/$ :failed)])]})
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn extract-command-event-signal
+  [m]
+  (cond
+    ;; commands
+    (contains? m :kixi.comms.command/key)
+    (select-keys m [:kixi.comms.command/key])
+    ;; events with `file-metadata-update-type` in the payload
+    (and (contains? m :kixi.comms.event/key)
+         (contains? m :kixi.comms.event/payload)
+         (contains? (:kixi.comms.event/payload m) :kixi.datastore.communication-specs/file-metadata-update-type))
+    (merge (select-keys m [:kixi.comms.event/key])
+           {:kixi.comms.event/payload (select-keys (:kixi.comms.event/payload m) [:kixi.datastore.communication-specs/file-metadata-update-type])})
+    ;; events
+    (contains? m :kixi.comms.event/key)
```

Comment:
No, it's cool, I intentionally left out any throwing simply because not all commands and events might fit into an activity, it's too early to say really.